apps/intellij/src/main/kotlin/dev/nx/console/generate/NxGeneratorSearchEvrywhereContributor.kt

```diff
@@ -0,0 +1,121 @@
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.util.Processor
+import dev.nx.console.generate.ui.DefaultNxGenerateUiFile
+import dev.nx.console.generate.ui.NxGeneratorListCellRenderer
+import dev.nx.console.models.NxGenerator
+import dev.nx.console.models.NxGeneratorOption
+import dev.nx.console.nxls.server.requests.NxGeneratorOptionsRequestOptions
+import dev.nx.console.services.NxlsService
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import javax.swing.ListCellRenderer
+import kotlinx.coroutines.runBlocking
+
+class NxGeneratorSearchEvrywhereContributor : SearchEverywhereContributorFactory<NxGenerator> {
+    override fun createContributor(
+        initEvent: AnActionEvent
+    ): SearchEverywhereContributor<NxGenerator> = NxGeneratorSEContributor(initEvent)
+}
+
+class NxGeneratorSEContributor(val event: AnActionEvent) :
+    SearchEverywhereContributor<NxGenerator> {
+
+    private val project = event.getRequiredData(CommonDataKeys.PROJECT)
+
+    private val generators = runBlocking { project.service<NxlsService>().generators() }
+
+    private val generatorToOptionsCache: ConcurrentMap<NxGenerator, List<NxGeneratorOption>> =
+        ConcurrentHashMap()
+
+    override fun getSearchProviderId(): String = javaClass.name
+
+    override fun getGroupName(): String = "Nx"
+
+    override fun getSortWeight(): Int = 0
+
+    override fun showInFindResults(): Boolean = true
+
+    override fun getElementsRenderer(): ListCellRenderer<in NxGenerator> =
+        NxGeneratorListCellRenderer(
+            myIcon = AllIcons.Actions.GeneratedFolder,
+            alternatingRowColors = false,
+        )
+
+    override fun getDataForItem(element: NxGenerator, dataId: String): Any? {
+        return null
+    }
+
+    override fun processSelectedItem(
+        selected: NxGenerator,
+        modifiers: Int,
+        searchText: String
+    ): Boolean {
+        val service = project.service<NxlsService>()
+        val generatorOptions: List<NxGeneratorOption>? =
+            generatorToOptionsCache.computeIfAbsent(selected) {
+                runBlocking {
+                    service.generatorOptions(
+                        NxGeneratorOptionsRequestOptions(it.data.collection, it.name, it.path)
+                    )
+                }
+            }
+
+        val path =
+            if (ActionPlaces.isPopupPlace(event.place)) {
+                event.dataContext.getData(CommonDataKeys.VIRTUAL_FILE)?.path
+            } else {
+                null
+            }
+
+        val generatorWithOptions = NxGenerator(selected, generatorOptions)
+
+        val generatorContext = runBlocking {
+            path?.let { service.generatorContextFromPath(generatorWithOptions, path) }
+        }
+
+        val virtualFile = DefaultNxGenerateUiFile("Generate", project)
```

Comment:
Since this is the same code as in the `Nx Generate (UI)` action, we could reuse it by moving it to the `NxGenerateService`

Reply:
I'm aware of this duplicate code, I will see if I can refactor it.

Reply:
I reused the openGenerateUi method for the moment